### PR TITLE
Item Collection Component

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Item Collection component implementation. A Item Collection is a wrapper component designed to display data as either a table or list of items dependent on the relative container size. 

@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-item-collection/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Terra Clinical Item Collection
+
+[![NPM version](http://img.shields.io/npm/v/terra-clinical-item-collection.svg)](https://www.npmjs.org/package/terra-clinical-item-collection)
+[![Build Status](https://travis-ci.org/cerner/terra-core.svg?branch=master)](https://travis-ci.org/cerner/terra-core)
+
+A Item Collection is a wrapper component designed to display data as either a table or list of items dependent on the relative container size. Initially data is displayed in a tabular format that flexes the rendered component, but as the container is resized and the indicated breakpoint is hit, the display changes to shows the data as a list of items. The terra-table and terra-clinical-item react components will be utilized for displaying the data in these two ways.
```
Comment (ryanthemanuel) [inline packages/terra-clinical-item-collection/README.md]: Nitpick, but should be An Item Collection

File: packages/terra-clinical-item-collection/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Ultimately, the Item Collection component component allows consumers to organize tabular data and ensure this data is readable as screen sizes become progressively smaller.
```
Comment (ryanthemanuel) [inline packages/terra-clinical-item-collection/README.md]: component is repeated here.

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  const { startAccessoryColumn, displayColumns, commentColumn, endAccessoryColumn } = tableColumns;
+
+  let startAccessoryHeader;
+  if (startAccessoryColumn) {
+    startAccessoryHeader = <TableHeaderCell columnType="accessory" key="start_accessory" />;
+  }
+
+  const displayHeaders = [];
+  if (displayColumns) {
+    for (let index = 0; index < displayColumns; index += 1) {
```
Comment (ryanthemanuel) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: Not a huge deal, but could we use: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach?v=control instead?

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    const displayContent = [];
+    if (displayColumns) {
+      for (let index = 0; index < displayColumns; index += 1) {
+        const contentKey = `display_${index + 1}`;
+        const content = row.displays[index] != null ? row.displays[index] : ' ';
+        displayContent[index] = (<Table.Cell content={content} key={contentKey} />);
+      }
+    }
+
+    let commentContent;
```
Comment (ryanthemanuel) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: 77 and 83 look like they're doing the same thing.  Do we think that it's worth making a common function here?

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx (hunk 3 above)
Comment (dkasper-was-taken) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: In javascript foreach is discouraged, as it has poor performance, i believe the eslint will even throw a warning if you use it.

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx (hunk 4 above)
Comment (dkasper-was-taken) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: +1

File: packages/terra-clinical-item-collection/src/_TableHeaderCell.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import './ItemCollection.scss';
```
Comment (dkasper-was-taken) [inline packages/terra-clinical-item-collection/src/_TableHeaderCell.jsx]: Does the _TableHeaderCell need to pull in this scss?

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import Table from 'terra-table';
+// eslint-disable-next-line import/no-named-as-default, import/no-named-as-default-member
+import TableHeaderCell from './_TableHeaderCell';
+import './ItemCollection.scss';
```
Comment (dkasper-was-taken) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: Same here, is this scss required.

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx (hunk 3 above)
Comment (ryanthemanuel) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: Hmm...  I was going off of https://github.com/airbnb/javascript#iterators-and-generators where forEach is "good" though not the "best"

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx (hunk 4 above)
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: Pulled logic out into `createTableCell`

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx (hunk 3 above)
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: I updated this to use the `.map` function

File: packages/terra-clinical-item-collection/src/_CreateTableView.jsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  let endAccessoryHeader;
+  if (endAccessoryColumn) {
+    endAccessoryHeader = <TableHeaderCell columnType="accessory" key="end_accessory" />;
+  }
+
+  return (
+    <Table.Header style={{ visibility: 'hidden' }} >
+      { startAccessoryHeader }
```
Comment (tbiethman) [inline packages/terra-clinical-item-collection/src/_CreateTableView.jsx]: Nitpicking, but the spaces around these braces are weird

**Comment by neilpfeiffer** [comment]
Tested: PR [#91]( https://github.com/cerner/terra-clinical/pull/91)

**Reviewed:**
- Follows initial implementation of the clinical item collection (formerly the CompactCard) and responsive behavior of switching between table and list layouts at a specified breakpoint.


**Pass** | Comments |
-----------| -----------| 
UX Review  | _pass for initial release – requires enhancement_ |


.
**Required Enhancement:** resolve with new issue
- [ ] Fix line-height/margin/padding/vertical-aligment of Displays when presented in the table view and when there is only one line of text so that all text “baselines” are aligned and icons and text all look centered vertically on same line.

**Comment by emilyrohrbough** [comment]
Address Issue #100

**Pull request merged**